webapp/src/ts/services/training-cards.service.ts

```diff
@@ -0,0 +1,124 @@
+import { Injectable } from '@angular/core';
+import { Store } from '@ngrx/store';
+
+import { XmlFormsService } from '@mm-services/xml-forms.service';
+import { TrainingCardsComponent } from '@mm-modals/training-cards/training-cards.component';
+import { DbService } from '@mm-services/db.service';
+import { GlobalActions } from '@mm-actions/global';
+import { ModalService } from '@mm-modals/mm-modal/mm-modal';
+import { SessionService } from '@mm-services/session.service';
+import { RouteSnapshotService } from '@mm-services/route-snapshot.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TrainingCardsService {
+  private globalActions;
+  private readonly TRAINING_PREFIX = 'training:'
+
+  constructor(
+    private store: Store,
+    private xmlFormsService: XmlFormsService,
+    private dbService: DbService,
+    private modalService: ModalService,
+    private sessionService: SessionService,
+    private routeSnapshotService: RouteSnapshotService,
+  ) {
+    this.globalActions = new GlobalActions(store);
+  }
+
+  private canOpenTraining() {
+    const routeSnapshot = this.routeSnapshotService.get();
+    return !this.sessionService.isDbAdmin() &&
+      (!document.getElementsByClassName('enketo').length ||
```

Comment:
This is another situation where we really don't want to rely on CSS. Can't we ask the enketo service if it's currently got a form rendered?

Reply:
currently we're not keeping an state of the form enketo is using. I can add something in the redux

Reply:
I tried a bunch of things (DOM query, CSS, state in reducers, etc) but the most stable and effective way so far has been to add a hideTraining property in the route’s data of pages where trainings shouldn’t display